diff --git a/node_modules/@volar/test-utils/index.js b/node_modules/@volar/test-utils/index.js
index 830c066..816c26b 100644
--- a/node_modules/@volar/test-utils/index.js
+++ b/node_modules/@volar/test-utils/index.js
@@ -17,6 +17,14 @@ function startLanguageServer(serverModule, cwd) {
         cwd,
         stdio: 'pipe',
     });
+    const logFilename = 'server.log';
+    const logFile = fs.createWriteStream
+        ? fs.createWriteStream(logFilename, { flags: 'w' })
+        : null;
+    if (logFile) {
+        childProcess.stdout.pipe(logFile);
+        childProcess.stderr.pipe(logFile);
+    }
     const connection = _.createProtocolConnection(childProcess.stdout, childProcess.stdin);
     const documentVersions = new Map();
     const openedDocuments = new Map();
@@ -153,8 +161,6 @@ function startLanguageServer(serverModule, cwd) {
                 textDocument: { uri },
                 position,
             });
-            // @volar/language-server only returns CompletionList
-            assert(!Array.isArray(result));
             return result;
         },
         sendCompletionResolveRequest(item) {
